assets/src/components/site-scan-context-provider/test/get-slugs-from-validation-results.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Internal dependencies
+ */
+import { getSlugsFromValidationResults } from '../get-slugs-from-validation-results';
+
+describe( 'getSlugsFromValidationResults', () => {
+	it( 'returns empty arrays if no validation results are passed', () => {
+		expect( getSlugsFromValidationResults() ).toMatchObject( { plugins: [], themes: [] } );
+	} );
+
+	it( 'returns plugin and theme slugs', () => {
+		const validationResult = [
+			{
+				sources: [
+					{ type: 'core', name: 'wp-includes' },
+					{ type: 'plugin', name: 'amp' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'plugin', name: 'gutenberg' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'core', name: 'wp-includes' },
+					{ type: 'plugin', name: 'jetpack' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'plugin', name: 'jetpack' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'plugin', name: 'foo-bar/foo-bar.php' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'theme', name: 'twentytwenty' },
+					{ type: 'core', name: 'wp-includes' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'core', name: 'wp-includes' },
+				],
+			},
+			{
+				sources: [
+					{ type: 'theme', name: 'twentytwenty' },
+					{ type: 'core', name: 'wp-includes' },
+				],
+			},
+		];
+
+		const slugs = getSlugsFromValidationResults( validationResult );
+
+		expect( slugs.plugins ).toStrictEqual( [ 'jetpack', 'foo-bar' ] );
```

Comment:
In this case, `gutenberg` should actually be included in the list because there was a validation error that has Gutenberg as the only source _for that one validation error_.

Reply:
Yup. I'll write an additional test case that will check if `gutenberg` is removed if there are more plugin sources for the same validation error.
```suggestion
		expect( slugs.plugins ).toStrictEqual( [ 'gutenberg', 'jetpack', 'foo-bar' ] );
```

Reply:
The additional test has been added in 7d2f1d1.